src/servers/Scripts/quest/classquest/ARC/ClsArc000.cpp

```diff
@@ -0,0 +1,180 @@
+#include <Actor/Player.h>
+#include <Event/EventHelper.h>
+#include <ScriptObject.h>
+
+// Quest Script: ClsArc000_00021
+// Quest Name: Way of the Archer
+// Quest ID: 65557
+// Start NPC: 1000197
+// End NPC: 1000200
+
+class ClsArc000 : public EventScript
+{
+   private:
+      // Basic quest information 
+      // Quest vars / flags used
+      // GetQuestUI8AL
+      // GetQuestUI8BH
+      // GetQuestUI8BL
+
+      enum Sequence : uint8_t
+      {
+         Seq0 = 0,
+         Seq1 = 1,
+         Seq2 = 2,
+         SeqFinish = 255,
+      };
+
+      // Quest rewards 
+      static constexpr auto RewardExpFactor = 100;
+      static constexpr auto RewardItem = 4657;
+      static constexpr auto RewardItemCount = 5;
+
+      // Entities found in the script data of the quest
+      static constexpr auto Actor0 = 1000197;
+      static constexpr auto Actor1 = 1000200;
+      static constexpr auto Enemy0 = 37;
+      static constexpr auto Enemy1 = 49;
+      static constexpr auto Enemy2 = 47;
+      static constexpr auto LocFace0 = 604;
+      static constexpr auto LocFace1 = 605;
+      static constexpr auto LogmessageMonsterNotePageUnlock = 1008;
+      static constexpr auto Seq0Actor0 = 0;
+      static constexpr auto Seq1Actor1 = 1;
+      static constexpr auto Seq3Actor1 = 2;
+      static constexpr auto UnlockImageMonsterNote = 32;
+
+   public:
+      ClsArc000() : EventScript( 65557 )
+      {}; 
+      ~ClsArc000()
+      {}; 
+
+   void onTalk( uint32_t eventId, Entity::Player& player, uint64_t actorId ) override
+   {
+      auto actor = Event::mapEventActorToRealActor( static_cast< uint32_t >( actorId ) );
+
+      if( actor == Actor0 )
+      {
+         Scene00000( player );
+      }
+      if( actor == Actor1 && player.getQuestSeq( getId() ) == Seq1 )
+      {
+         Scene00001( player );
+      }
+      if( actor == Actor1 && player.getQuestSeq( getId() ) == SeqFinish )
+      {
+         Scene00002( player );
+      }
+   }
+
+   void onMobKill( Entity::Player& player, uint64_t npcId )
+   {
+      if( npcId != Enemy0 || Enemy1 || Enemy2 )
+         return;
+
+      auto currentKC37 = player.getQuestUI8AL( getId() ) + 1;
+      auto currentKC49 = player.getQuestUI8BH( getId() ) + 1;
+      auto currentKC47 = player.getQuestUI8BL( getId() ) + 1;
```

Comment:
this isn't correct, you increment each counter regardless of which mob was killed

Reply:
QuestBits are inside the lua script
```
    if L6_41 == A0_35.SEQ_2 then
      if A3_38 == A0_35.ENEMY0 then
        L7_42 = A1_36:GetQuestUI8AL(L5_40)
        if L7_42 >= 3 then
          return false
        end
        return 3 > A1_36:GetQuestUI8AL(L5_40)
      elseif A3_38 == A0_35.ENEMY1 then
        L7_42 = A1_36:GetQuestUI8BH(L5_40)
        if L7_42 >= 3 then
          return false
        end
        return 3 > A1_36:GetQuestUI8BH(L5_40)
      elseif A3_38 == A0_35.ENEMY2 then
        L7_42 = A1_36:GetQuestUI8BL(L5_40)
        if L7_42 >= 3 then
          return false
        end
        return 3 > A1_36:GetQuestUI8BL(L5_40)
      end
    end
```

Reply:
which isn't what you're doing, you're incrementing the kill count for each one regardless of the npc killed

you also don't check the quest sequence either